### PR TITLE
[webgui] introduce new default webgui display kind

### PR DESCRIPTION
**Pull request by linev**
Enabled when configured with `root --web` or `root --web=on`
It tries to use local displays like CEF or Qt6 or Qt5 first. 
If not available, try to use detected "native" browsers like Chrome or Firefox. 
All these display kinds provides comfortable window management which are very like to normal windows.
Only if none of this will work default system web browser will be started - as it was up to now.

Extend TROOT::SetWebDisplay() docu and `root --help` output

### CODE REVIEW

**Review by couet — APPROVED** [review]
LGTM

File: core/base/src/TROOT.cxx [hunk 1]
```diff
@@ -2776,6 +2776,13 @@ void TROOT::SetMacroPath(const char *newpath)
 ///  - "firefox": select Mozilla Firefox browser for interactive web display
 ///  - "chrome": select Google Chrome browser for interactive web display
 ///  - "edge": select Microsoft Edge browser for interactive web display
+///  - "native": select one of the natively-supported web browsers firefox/chrome/edge for interactive web display
+///  - "qt5": uses QWebEngine from Qt5, no real http server started (require qt5web component build for ROOT)
```
Comment (dpiparo) [inline core/base/src/TROOT.cxx:2780]: require -> requires ROOT's qt5web component

File: core/base/src/TROOT.cxx [hunk 2]
```diff
@@ -2776,6 +2776,13 @@ void TROOT::SetMacroPath(const char *newpath)
...
+///  - "qt6": uses QWebEngine from Qt6, no real http server started (require qt6web component build for ROOT)
```
Comment (dpiparo) [inline core/base/src/TROOT.cxx:2781]: see above.

File: core/base/src/TROOT.cxx [hunk 3]
```diff
@@ -2776,6 +2776,13 @@ void TROOT::SetMacroPath(const char *newpath)
...
+///  - "cef": uses Chromium Embded Framework, no real http server started (require cefweb component build for ROOT)
```
Comment (dpiparo) [inline core/base/src/TROOT.cxx:2782]: embedded, see above for the "require..."

File: core/base/src/TROOT.cxx [hunk 4]
```diff
@@ -2776,6 +2776,13 @@ void TROOT::SetMacroPath(const char *newpath)
...
+///  - "local": select on of available local (without http server) engines like qt5/qt6/cef
+///  - "default": system default web browser, invoked with xdg-open on Linux or open on Windows
```
Comment (dpiparo) [inline core/base/src/TROOT.cxx:2784]: what about macos?

File: core/base/src/root-argparse.py [hunk 5]
```diff
@@ -18,8 +18,9 @@ def get_argparse():
 	parser.add_argument('-h','-?', '--help', help='Show summary of options')
 	parser.add_argument('--version', help='Show the ROOT version')
 	parser.add_argument('--notebook', help='Execute ROOT notebook')
-	parser.add_argument('--web', help='Display graphics in a default web browser')
-	parser.add_argument('--web=<browser>', help='Display graphics in specified web browser')
+	parser.add_argument('--web', help='Use web-based display for graphics, browser, geometry')
+	parser.add_argument('--web=<kind>', help='Use specified kind for web-based display like chrome, firefox, qt5\nMore available options see in TROOT::SetWebDisplay() documentation')
```
Comment (dpiparo) [inline core/base/src/root-argparse.py:22]: "Use the specified web-based display, such as .... For more options, see the documentation of TROOT::SetWebDisplay()"

Comment (dpiparo) [inline core/base/src/root-argparse.py:22]: kind --> type?

File: gui/webdisplay/inc/ROOT/RWebDisplayArgs.hxx [hunk 6]
```diff
@@ -48,6 +48,7 @@ public:
       kServer,   ///< indicates that ROOT runs as server and just printouts window URL, browser should be started by the user
       kEmbedded, ///< window will be embedded into other, no extra browser need to be started
       kOff,      ///< disable web display, do not start any browser
+      kOn,       ///< enabled web display, first try use embed displays like Qt or CEF, then native browsers and at the end default system browser
```
Comment (dpiparo) [inline gui/webdisplay/inc/ROOT/RWebDisplayArgs.hxx:51]: "web display enable" ?

File: gui/webdisplay/src/RWebDisplayArgs.cxx [hunk 7]
```diff
@@ -143,15 +143,17 @@ bool RWebDisplayArgs::SetPosAsStr(const std::string &str)
...
 ///
-///      chrome - use Google Chrome web browser, supports headless mode from v60, default
-///     firefox - use Mozilla Firefox browser, supports headless mode from v57
-///      native - (or empty string) either chrome or firefox, only these browsers support batch (headless) mode
-///     browser - default system web-browser, no batch mode
-///      safari - Safari browser on Mac
+///      chrome - use Google Chrome web browser
+///     firefox - use Mozilla Firefox browser
+///        edge - use Edge web browser (Windows only)
+///      native - either chrome or firefox, only these browsers support batch (headless) mode
```
Comment (dpiparo) [inline gui/webdisplay/src/RWebDisplayArgs.cxx:149]: what happens in N years when edge changes name? Can we be more future proof and write "(curently edge)"?

**Review by dpiparo — APPROVED** [review]
Just cosmetics, but that are important.

**Comment by enirolf** [comment]
Hi @linev, it looks like this PR did not pass the `clang-format` check, which is now affecting new PRs (e.g., #15084). Could you fix the formatting? Thanks!

**Comment by linev** [comment]
@enirolf Strange that changes in webgui has effect on ntuple PRs.

ROOT source code by 95% do not pass clang-format test - especially header files.
Therefore I often ignore this warning during PRs - otherwise header files formatting will be confusing.

**Pull request merged**